File: eureka-consumer/src/main/kotlin/com/bbz/learn/eurekaconsumer/service/HelloRemoteService.java
//package com.bbz.learn.eurekaconsumer.service;
//
//
//import org.springframework.cloud.openfeign.FeignClient;
//import org.springframework.stereotype.Service;
//import org.springframework.web.bind.annotation.RequestMapping;
//import org.springframework.web.bind.annotation.RequestParam;
//
//@Service
//@FeignClient(value = "demo-client",fallback = HelloRemoteServiceHystric.class)
//public interface HelloRemoteService {
//    @RequestMapping(value = "/producerHello")
//     String sayHello(@RequestParam("name") String name);
//}

File: eureka-consumer/src/main/kotlin/com/bbz/learn/eurekaconsumer/service/HelloRemoteService.kt
package com.bbz.learn.eurekaconsumer.service

import org.springframework.cloud.openfeign.FeignClient
import org.springframework.stereotype.Component
import org.springframework.stereotype.Service
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RequestMapping



@Service
@FeignClient(name = "demo-client",fallback = HelloRemoteServiceHystric::class)
open interface HelloRemoteService {

    @RequestMapping(value = ["/producerHello"])
    fun sayHello(@RequestParam("name") name: String): String
}

@Component
class HelloRemoteServiceHystric : HelloRemoteService {
    override fun sayHello(name: String): String {
        return "sorry$name"
    }
}